mmpose/models/heads/vis_heads/vis_head.py

```diff
@@ -0,0 +1,213 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Sequence, Tuple, Union
+
+import torch
+from torch import Tensor, nn
+
+from mmpose.models.utils.tta import flip_visibility
+from mmpose.registry import MODELS
+from mmpose.utils.tensor_utils import to_numpy
+from mmpose.utils.typing import (ConfigType, InstanceList, OptConfigType,
+                                 OptSampleList, Predictions)
+from ..base_head import BaseHead
+
+
+@MODELS.register_module()
+class VisibilityPredictionHead(BaseHead):
+    """
+    Args:
+        in_channels (int | sequence[int]): Number of input channels
+        num_joints (int): Number of joints
+        loss (Config): Config for visibility loss. Defaults to use
+            :class:`BCELoss`
+        pose_cfg (Config): Config to construct keypoints prediction head
+        init_cfg (Config, optional): Config to control the initialization. See
+            :attr:`default_init_cfg` for default settings
+    """
+
+    def __init__(self,
+                 in_channels: Union[int, Sequence[int]],
+                 num_joints: int,
+                 pose_cfg: ConfigType,
+                 loss: ConfigType = dict(
+                     type='BCELoss', use_target_weight=True),
+                 init_cfg: OptConfigType = None):
+
+        if init_cfg is None:
+            init_cfg = self.default_init_cfg
+
+        super().__init__(init_cfg)
+
+        self.in_channels = in_channels
+        self.num_joints = num_joints
+        self.loss_module = MODELS.build(loss)
+
+        self.pose_head = MODELS.build(pose_cfg)
+        self.pose_cfg = pose_cfg
+
+        self.vis_head = nn.Sequential(
+            nn.AdaptiveAvgPool2d(1), nn.Flatten(),
+            nn.Linear(self.in_channels, self.num_joints), nn.Sigmoid())
```

Comment:
I suggest adding a `use_sigmoid: bool = False` in `__init__()` to determine wether to use `sigmoid` in `vis_head`, because Pytorch officially recommand to use `BCEWithLogitsLoss` instead of `Sigmoid + BCELoss`, which is "more numerically stable than using a plain Sigmoid followed by a BCELoss as, by combining the operations into one layer, we take advantage of the log-sum-exp trick for numerical stability." [details](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html)